Add type annotations to `lib/utils/window.ts`

## Context
Continuing with the [conversion to Typescript](https://github.com/thoughtbot/superglue/issues/43). There's 1 function in `lib/utils/window.ts` that needs to be converted to typescript. 


## Issue
`npm run lint` is showing the following warnings:

```
  1:8   warning  Missing return type on function        @typescript-eslint/explicit-module-boundary-types
  1:30  warning  Argument 'prevAssets' should be typed  @typescript-eslint/explicit-module-boundary-types
  1:42  warning  Argument 'newAssets' should be typed   @typescript-eslint/explicit-module-boundary-types
```

Lets go ahead and convert this one function. This is a easy win. Both `prevAssets` and `nextAssets` are both `string[]`.

--- a/superglue/lib/utils/window.ts
+++ b/superglue/lib/utils/window.ts
@@ -1,4 +1,7 @@
-export function needsRefresh(prevAssets, newAssets) {
+export function needsRefresh(
+  prevAssets: string[],
+  newAssets: string[]
+): boolean {
   if (prevAssets && newAssets) {
     const hasNewAssets = !newAssets.every((asset) =>
       prevAssets.includes(asset)

